comp110-worksheet-6/OxoBoard.cs

```diff
@@ -10,39 +10,76 @@ public enum Mark { None, O, X };
 
 	public class OxoBoard
 	{
-		// Constructor. Perform any necessary data initialisation here.
+        // Constructor. Perform any necessary data initialisation here.
+        Mark[,] Board;
 		// Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
-		public OxoBoard(/* int width = 3, int height = 3, int inARow = 3 */)
+		public OxoBoard (int width = 3, int height = 3, int inARow = 3)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            Board = new Mark[width, height];
 		}
 
 		// Return the contents of the specified square.
 		public Mark GetSquare(int x, int y)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            return Board[x, y];
 		}
 
 		// If the specified square is currently empty, fill it with mark and return true.
 		// If the square is not empty, leave it as-is and return False.
 		public bool SetSquare(int x, int y, Mark mark)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            if (x < Board.GetLongLength(0) && y < Board.GetLongLength(1) && x > -1 && y > -1)
+            {
+                if (Board[x, y] == Mark.None)
+                {
+                    Board[x, y] = mark;
+                }
+            }
+            return false;
 		}
 
 		// If there are still empty squares on the board, return false.
 		// If there are no empty squares, return true.
 		public bool IsBoardFull()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            for (int A = 0; A < Board.GetLongLength(0);)
+            {
+                A + 1;
+                for (int B = 0; B < Board.GetLongLength(1);)
+                {
+                    B + 1;
+                    return false;
+                }
+            }
+           return true;
 		}
 
-		// If a player has three in a row, return Mark.O or Mark.X depending on which player.
-		// Otherwise, return Mark.None.
-		public Mark GetWinner()
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+        // If a player has three in a row, return Mark.O or Mark.X depending on which player.
+        // Otherwise, return Mark.None.
+        public Mark GetWinner()
+        {
+            for (int Yaxis = 0; Yaxis < Board.GetLongLength(1); )
+        {
+                Yaxis + 1;
+                if (Board[0, Yaxis] == Board[1, Yaxis] && Board[1, Yaxis] == Board[2, Yaxis])
+                {
+                    if (GetSquare(0, Yaxis) != Mark.None)
+                    {
+                        return GetSquare(0 ,Yaxis);
+                    }
+                }
+            }
```

Comment:
It looks like you're thinking along the right lines here -- just some revision of how to use the `for` loop in C# should put you on the right track